apps/etterlatte-samordning-vedtak/src/main/kotlin/no/nav/etterlatte/samordning/vedtak/LoggingPlugins.kt

```diff
@@ -0,0 +1,113 @@
+package no.nav.etterlatte.samordning.vedtak
+
+import io.ktor.server.application.ApplicationCall
+import io.ktor.server.application.ApplicationCallPipeline
+import io.ktor.server.application.Hook
+import io.ktor.server.application.RouteScopedPlugin
+import io.ktor.server.application.call
+import io.ktor.server.application.createApplicationPlugin
+import io.ktor.server.application.createRouteScopedPlugin
+import io.ktor.server.application.hooks.ResponseSent
+import io.ktor.server.request.httpMethod
+import io.ktor.server.request.path
+import io.ktor.server.request.uri
+import io.ktor.util.AttributeKey
+import io.ktor.util.logging.KtorSimpleLogger
+import io.ktor.util.pipeline.PipelinePhase
+import net.logstash.logback.marker.Markers
+import no.nav.etterlatte.libs.ktor.PluginConfiguration
+import no.nav.etterlatte.libs.ktor.brukerTokenInfo
+import no.nav.etterlatte.token.Saksbehandler
+import no.nav.etterlatte.token.Systembruker
+import org.slf4j.MDC
+
+internal val LOGGER = KtorSimpleLogger("no.nav.etterlatte.samordning.requestLogger")
+
+private val userAttribute = AttributeKey<String>("user")
+private val startTimeAttribute = AttributeKey<Long>("starttime")
+private val loggingPerformed = AttributeKey<Boolean>("requestLoggingPerformed")
+
+private object UserIdMdcHook : Hook<suspend (ApplicationCall) -> Unit> {
+    private val UserIdMdcHook: PipelinePhase = PipelinePhase("UserIdMdc")
+    private val AuthenticatePhase: PipelinePhase = PipelinePhase("Authenticate")
+
+    override fun install(
+        pipeline: ApplicationCallPipeline,
+        handler: suspend (ApplicationCall) -> Unit,
+    ) {
+        // Se Tilgangsstyring.kt
+        pipeline.insertPhaseAfter(ApplicationCallPipeline.Plugins, AuthenticatePhase)
+        pipeline.insertPhaseAfter(AuthenticatePhase, UserIdMdcHook)
+        pipeline.intercept(UserIdMdcHook) { handler(call) }
+    }
+}
+
+val userIdMdcPlugin: RouteScopedPlugin<PluginConfiguration> =
+    createRouteScopedPlugin(
+        name = "UserIdMdcPlugin",
+        createConfiguration = ::PluginConfiguration,
+    ) {
+        on(UserIdMdcHook) { call ->
+            val user =
+                if (call.request.uri.contains("pensjon")) {
+                    when (val bruker = call.brukerTokenInfo) {
+                        is Systembruker -> bruker.sub
```

Comment:
Er dette personvernmessig ok? Har vært borti før at logging av saksbehandler-identer kommer med strenge regler... Hvordan skal identen brukes?

Reply:
For dette caset så er det vel et spm inntil videre hvorvidt det er saksbehandler eller systemet, dvs "pensjon-pen" som blir benyttet. Men saksbehandleridenter logges en masse og benyttes. Aldri hørt om at det er et problem. Snarere tvert imot så synes jeg vi i Gjenny har fullstendig mangel på den slags mtp sporing/analyse av feilsituasjoner.

Når det er sagt så er det egentlig ikke interessant for dashboardet her å vite eksplisitte saksbehandleridenter, så jeg kan oppdatere til bare "Saksbehandler". Like greit det, så kan man bedre gruppere.